src/components/general/TextInput/index.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import styles from './styles.less';
+import classNames from 'classnames';
+
+type TextInputProps = {
+    disabled?: boolean;
+    error?: boolean;
+    errorMessage?: string;
+    isRequired?: boolean;
+    placeholder?: string;
+    label?: string;
+    onChange: (newValue: string) => void;
+    value?: string;
+    icon?: any;
+    onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
+    onIconAction?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+};
+
+const TextInput = React.forwardRef<HTMLDivElement | null, React.PropsWithChildren<TextInputProps>>(
+    (
+        {
+            disabled = false,
+            error,
+            errorMessage,
+            isRequired = false,
+            placeholder = '',
+            label,
+            onChange,
+            value = '',
+            icon,
+            onBlur,
+            onIconAction,
+            ...props
+        },
+        ref
+    ) => {
+        const inputRef = React.useRef<HTMLInputElement>(null);
+        const [focus, setFocus] = React.useState(false);
+        const inputContainerRef = ref as React.RefObject<HTMLDivElement | null>;
```

Comment:
As a user of this component I might expect the forwarded ref to point to the input field, not the wrapping container?

Reply:
Ref is now pointing to the input element